skopt/optimizer/optimizer.py

```diff
@@ -520,6 +520,31 @@ def tell(self, x, y, fit=True):
         return create_result(self.Xi, self.yi, self.space, self.rng,
                              models=self.models)
 
+    def _check_y_is_valid(self, x, y):
+        """Checks if y is valid (with respect to x in some cases)."""
```

Comment:
How could we reformulate it so that the doc string tells us something the function name doesn't already tell us?

Don't have a good suggestion right now for wording. Basically outline what kind of things we are checking ... without repeating the whole code. Maybe something like "Check if the shape and types of `x` and `y` are consistent."?

Reply:
I fully agree. This was added because i was asking myself at some point, why i need `x` to validate `y`.

"Check if the shape and types of x and y are consistent." sound good to me.

Reply:
Or rather "shape*s* and types"?